src/main/scala/devices/spi/SPIPhysical.scala

```diff
@@ -59,11 +98,39 @@ class SPIPhysical(c: SPIParamsBase) extends Module {
     Mux(fmt.endian === SPIEndian.MSB, data, Cat(data.toBools))
 
   val rxd = Cat(io.port.dq.reverse.map(_.i))
-  val samples = Seq(rxd(1), rxd(1, 0), rxd)
+  val rxd_delayed = Vec(Seq.fill(io.port.dq.size)(false.B))
+  //0.U(io.port.dq.size.W).toBools
+  //Adding fine-granularity delay buffers on the received data
+
+  val fine_grain_delay_0 = Module(new BlackBoxDelayBuffer())
```

Comment:
could we make this a list of these:

`val fine_grain_delay = Seq.fill(4){Module(new BlackBoxDelayBuffer()}`

Reply:
Sure, I didn't know if such thing (seq.fill) existed in chisel.

Reply:
Right, it's just Scala sequence, nothing Chisel specific, one of the benefits of using Scala